docs/get_started/chat_agent.md

```diff
@@ -0,0 +1,206 @@
+    ```python
+    agent.init_messages()
+    print(agent.stored_messages)
+    >>> [<BaseMessage object>, ...]
+    ```
+
+7. **Updating Stored Messages**:
+    The `update_messages` method allows you to add a new message to the stored messages and retrieve the updated list.
+    ```python
+    updated_messages = agent.update_messages(role="user", message=new_message)
+    print(updated_messages)
+    >>> [<BaseMessage object>, ...]
+    ```
+
+8. **Submitting an External Message as an Assistant Response**:
+    The `submit_message` method allows you to provide an external message as if it were a response from the chat LLM.
+    ```python
+    external_message = BaseMessage(...)
+    agent.submit_message(external_message)
+    ```
+
+9. **Performing a Chat Session Step**:
+    The `step` method lets you generate a response for a given input message and provides details of the chat session. 
+    ```python
+    input_msg = BaseMessage(...)
+    response = agent.step(input_message=input_msg)
+    print(response.output_messages)
+    >>> [<BaseMessage content="Generated response">]
+    ```
+
+10. **Preprocessing Messages for OpenAI Input**:
+    The `preprocess_messages` method truncates the list of messages if needed, converts them to OpenAI's input format, and calculates the token count.
+    ```python
+    chat_records = [...]
+    openai_msgs, tokens = agent.preprocess_messages(messages=chat_records)
+    print(tokens)
+    >>> 150
+    ```
+
+11. **Validating Model's Response**:
+    The `validate_model_response` method ensures that the response from the model is in the expected format.
+    ```python
+    model_response = {...}
+    agent.validate_model_response(response=model_response)
+    ```
+
+12. **Handling Batch Response from Model**:
+    The `handle_batch_response` method processes the batch response from the model and returns the chat messages, finish reasons, and other relevant details.
+    ```python
+    batch_response = {...}
+    messages, reasons, usage, response_id = agent.handle_batch_response(response=batch_response)
+    print(messages)
+    >>> [<BaseMessage content="Message 1">, <BaseMessage content="Message 2">]
+    ```
+
+13. **Handling Stream Response from Model**:
+    The `handle_stream_response` method processes the stream response from the model and returns the chat messages, finish reasons, and other relevant details.
+    ```python
+    stream_response = {...}
+    prompt_tokens_count = 100
+    messages, reasons, usage, response_id = agent.handle_stream_response(response=stream_response, prompt_tokens=prompt_tokens_count)
+    print(messages)
+    >>> [<BaseMessage content="Streamed Message 1">, <BaseMessage content="Streamed Message 2">]
+    ```
+
+14. **Handling Token Exceedance**:
+    The `step_token_exceed` method returns a response containing the number of tokens and information about called functions when the number of tokens exceeds the set limit.
```

Comment:
Actually it should be a private function. Could you please just remove it?

Reply:
> Actually it should be a private function. Could you please just remove it?

done